### PR TITLE
Add navigation component, and add basic state management

### PR DESCRIPTION
**Pull request by rohannelson**
Setup persistent state management with @nanostores/persistent, implemented this for setup-page (with the omission of the background image field), and setup basic page navigation (will likely need to be revised to be more modular).

### CODE REVIEW

**Comment by jasononeil** [comment]
In testing this it seems to be storing state on the server - I see the "In store listener" logs in the terminal not the browser console, and the name is persisted even across different browsers and private/incognito windows.

**Comment by rohannelson** [comment]
Got it working on the client side using clientLoader and clientAction (have tested and state is separate across a general and private/incognito session).
I changed store.ts to store.client.ts to indicate that it shouldn't be bundled on the server. Turns out this isn't necessary when not using a server side Action in combination with clientAction (which I did as a transitional thing trying to get it to work), but I'll leave it there for now as when using both action types it was causing problems.

File: app/store/page.client.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { persistentAtom } from "@nanostores/persistent";
+
+export type Page = {
+  name?: string;
+  background?: string;
+  weekStart?: string;
```
Comment (jasononeil) [inline app/store/page.client.ts:6]: I won't ask you to do it in this PR, but changing this from `string` to `"sunday" | "monday"` is a way to get more benefit from typescript, ensuring you know exactly what kind of strings you'll be working with

**Review by jasononeil — APPROVED** [review]

**Review by jasononeil — APPROVED** [review]

File: app/store/page.client.ts (hunk 1 above)
Comment (rohannelson) [reply app/store/page.client.ts:6]: Makes sense, I had a go at this but was having trouble with type narrowing in setup-page.tsx so I'm going to leave it for now and come back later.

Comment (rohannelson) [reply app/store/page.client.ts:6]: Never mind - got it. Had to add a return type.

**Pull request merged**